pkg/config/validation/validation.go

```diff
@@ -1129,33 +1129,45 @@ func validateLoadBalancer(settings *networking.LoadBalancerSettings) (errs error
 	return
 }
 
-func validateTLS(settings *networking.ClientTLSSettings) (errs error) {
+func validateTLS(settings *networking.ClientTLSSettings) Validation {
+	v := Validation{}
 	if settings == nil {
-		return
+		return v
 	}
 
 	if (settings.Mode == networking.ClientTLSSettings_SIMPLE || settings.Mode == networking.ClientTLSSettings_MUTUAL) &&
 		settings.CredentialName != "" {
 		if settings.ClientCertificate != "" || settings.CaCertificates != "" || settings.PrivateKey != "" {
-			errs = appendErrors(errs,
+			v = appendValidation(v,
 				fmt.Errorf("cannot specify client certificates or CA certificate If credentialName is set"))
 		}
 
 		// If tls mode is SIMPLE or MUTUAL, and CredentialName is specified, credentials are fetched
 		// remotely. ServerCertificate and CaCertificates fields are not required.
-		return
+		return v
 	}
 
 	if settings.Mode == networking.ClientTLSSettings_MUTUAL {
 		if settings.ClientCertificate == "" {
-			errs = appendErrors(errs, fmt.Errorf("client certificate required for mutual tls"))
+			v = appendValidation(v, fmt.Errorf("client certificate required for mutual tls"))
 		}
 		if settings.PrivateKey == "" {
-			errs = appendErrors(errs, fmt.Errorf("private key required for mutual tls"))
+			v = appendValidation(v, fmt.Errorf("private key required for mutual tls"))
+		}
+		if settings.CaCertificates == "" {
+			v = appendValidation(v, WrapWarning(fmt.Errorf("ca certificate required for mutual tls,  it will fail in release 1.12")))
 		}
 	}
 
-	return
+	if settings.Mode == networking.ClientTLSSettings_ISTIO_MUTUAL {
+		if settings.CredentialName != "" || settings.ClientCertificate != "" ||
+			settings.CaCertificates != "" || settings.PrivateKey != "" {
+			v = appendValidation(v, WrapWarning(fmt.Errorf("cannot specify privateKey, clientCertificate, caCertificates or"+
+				" credentialName for ISTIO_MUTUAL tls, it will fail in release 1.12")))
```

Comment:
```suggestion
				" credentialName for ISTIO_MUTUAL tls")))
```

Reply:
don't we need to give this note? Then we can strictly validate it.

Reply:
As mentioned above, we cannot ever strictly validate it. Its a breaking API change.

Reply:
I would suggest doing this like removing an API field or a whole API